gadgets/trace_exec/program.bpf.c

```diff
@@ -252,4 +253,31 @@ int ig_execve_x(struct syscall_trace_exit *ctx)
 	return 0;
 }
 
+SEC("kprobe/security_bprm_check")
+int BPF_KPROBE(security_bprm_check, struct linux_binprm *bprm)
+{
+	if (!paths)
+		return 0;
+
+	u32 pid = (u32)bpf_get_current_pid_tgid();
+	struct event *event;
+	char *file;
+	struct path f_path;
+
+	event = bpf_map_lookup_elem(&execs, &pid);
+	if (!event)
+		return 0;
+
+	// security_bprm_check is called repeatedly following the shebang
+	// Only get the first call.
+	if (event->file[0] != '\0')
+		return 0;
+
+	f_path = BPF_CORE_READ(bprm, file, f_path);
```

Comment:
Can we get `f_inode` from `bprm->file` and use the `has_upper_layer` function to improve the limitations around upper layer in the scenario of a script?

Doesn't have to be in this PR, just a thought

Reply:
Opened this for tracking [[RFE] Improve trace exec upper layer limitation in case of a script · Issue #3838 · inspektor-gadget/inspektor-gadget](https://github.com/inspektor-gadget/inspektor-gadget/issues/3838)